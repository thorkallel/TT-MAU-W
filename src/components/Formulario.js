import React, { useEffect, useState, useContext } from 'react';

//Componentes
import Error from './Error';
import useCriptomoneda from '../hooks/useCriptomoneda';

// Context
import criptoContext from '../context/cripto/criptoContext';

const Formulario = () => {

	// Importar Context state de cripto
	const CriptoContext = useContext(criptoContext);
	const {
			datacripto,
			loadstate,

			fGetDatacripto,
			fSaveCripto
	} = CriptoContext;

	// state del listado de cripto
	const [ listacripto, guardarcripto ] = useState([]);
	const [ error, guardarError ] = useState(false);

	// utilizar useCriptomoneda
	const [criptomoneda, SelectCripto] = useCriptomoneda('Elige tu moneda','' , listacripto);

	// Ejecutar a la API
	useEffect(() => {

		if(loadstate) {
			fGetDatacripto();
		}

		guardarcripto(datacripto);

	}, [loadstate])


 	// cuando el usuario hace submit
	 const cotizarMoneda = e => {

		 e.preventDefault();

		 if( criptomoneda === '') {
			guardarError(true);
			return;
		}

		 // Validar que se haya seleccionado una criptomoneda
		 guardarError(false);

		 // pasar los datos al componente principal
		 fSaveCripto(criptomoneda);
	 }

	return (
		<form
			onSubmit={cotizarMoneda}
			className="content--form"
		>

			{ error ? <Error mensaje="Por favor seleccione una criptomoneda" /> : null }

			<SelectCripto />

			<input
				type="submit"
				value="Calcular"
			></input>
		</form>
	 );
}

export default Formulario;