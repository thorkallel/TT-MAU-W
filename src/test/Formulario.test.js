import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";

import CriptoState from "../context/cripto/criptoState";
import CotizacionState from "../context/cotizacion/cotizacionState";

import Formulario from "../components/Formulario";

describe("Pruebas en <Formulario />", () => {
  test("debe mostrar <Formulario /> correctamente", () => {

    const wrapper = shallow(
      <CriptoState>
        <CotizacionState>
          <Formulario />
        </CotizacionState>
      </CriptoState>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
