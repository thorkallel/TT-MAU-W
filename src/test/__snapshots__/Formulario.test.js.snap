// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Formulario /> debe mostrar <Formulario /> correctamente 1`] = `
<ContextProvider
  value={
    Object {
      "criptomoneda": "",
      "datacripto": Array [],
      "fGetDatacripto": [Function],
      "fSaveCripto": [Function],
      "loadstate": true,
    }
  }
>
  <CotizacionState>
    <Formulario />
  </CotizacionState>
</ContextProvider>
`;
